### PR TITLE
NOS4 - Change home feed algorithm

### PR DESCRIPTION
**Pull request by cjorgensen**
- Adds a Profile class, which helps organize event requests
- Update EventProcessor to utilize Profile
- Changes Filter to use strings instead of Author class for easier use
- NOTE: The home feed Core Data fetch will need to be updated if someone adds a way to download texts from outside the Profile's follows (ie Search, Global feed)

### CODE REVIEW

**Review by shanecb — COMMENTED** [review]
This looks good. My only concern is that there's no way to follow anyone yet, so if we merge this we won't be able to see anything on the home feed for testing purposes until we add the ability to follow people, right?

File: Nos/Models/Nos.xcdatamodeld/Nos.xcdatamodel/contents [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <model type="com.apple.IDECoreDataModeler.DataModel" documentVersion="1.0" lastSavedToolsVersion="21513" systemVersion="21G115" minimumToolsVersion="Automatic" sourceLanguage="Swift" userDefinedModelVersionIdentifier="">
     <entity name="Author" representedClassName=".Author" syncable="YES" codeGenerationType="category">
         <attribute name="about" optional="YES" attributeType="String"/>
+        <attribute name="display_name" optional="YES" attributeType="String"/>
```
Comment (shanecb) [inline Nos/Models/Nos.xcdatamodeld/Nos.xcdatamodel/contents:5]: any reason this isn't camel case?

Comment (cjorgensen) [reply Nos/Models/Nos.xcdatamodeld/Nos.xcdatamodel/contents:5]: I was matching the JSON. I think we could get around that by using CodingKeys in the MetadataEventJSON class, then we could use a name without an underscore.

**Review by shanecb — APPROVED** [review]

**Review by mplorentz — APPROVED** [review]
This looks good! I plugged in my main identity and it correctly is showing notes from the people I follow! 

I noticed that it doesn't seem to be fetching the author information as reliably as it was before. Although I'm not sure if that's caused by this PR or by the relays I'm using or something else. I think we can address that in a future PR.

I think it would be nice to rename `Profile` to something more like `LoggedInProfile` or `CurrentUser` or something, because just `Profile` sounds a lot like `Author`.

**Pull request merged**